__init__.py

```diff
@@ -289,1510 +87,511 @@ def __init__(self):
-        self.speak_dialog('sunrise', {'time': spoken_time})
-
-    # Handle: When is the sunset?
-    @intent_handler(IntentBuilder("").one_of("Query", "When")
-                    .optionally("Location").require("Sunset").build())
+        # Handle: When is the sunrise?
+        intent_data = WeatherIntent(message, self.lang)
+        weather = self._get_weather(intent_data)
+        if weather is not None:
+            intent_weather = weather.get_weather_for_intent(intent_data)
+            dialog = WeatherDialog(
+                intent_weather, self.weather_config, intent_data
+            )
+            dialog.build_sunrise_dialog()
+            self._speak_weather(dialog)
+
+    @intent_handler(
+        IntentBuilder("").one_of("Query", "When").optionally("Location")
+        .require("Sunset")
+    )
     def handle_sunset(self, message):
-        report = self.__initialize_report(message)
-
-        when, _ = self.__extract_datetime(message.data.get('utterance'))
-        today, _ = self.__extract_datetime("today")
-        if when is None or when.date() == today.date():
-            weather = self.owm.weather_at_place(
-                report['full_location'],
-                report['lat'],
-                report['lon']).get_weather()
-
-            if weather is None:
-                self.__report_no_data('weather')
-                return
+        # Handle: When is the sunset?
+        intent_data = WeatherIntent(message, self.lang)
+        weather = self._get_weather(intent_data)
+        if weather is not None:
+            intent_weather = weather.get_weather_for_intent(intent_data)
+            dialog = WeatherDialog(
+                intent_weather, self.weather_config, intent_data
+            )
+            dialog.build_sunset_dialog()
+            self._speak_weather(dialog)
+
+    def _report_current_weather(self, message):
+        intent_data = self._get_intent_data(message)
+        weather = self._get_weather(intent_data)
+        if weather is not None:
+            dialog = WeatherDialog(
+                weather.current, self.weather_config, intent_data
+            )
+            dialog.build_current_weather_dialog()
+            self._display_current_weather(weather)
+            self._speak_weather(dialog)
+            dialog.build_high_low_temperature_dialog()
+            self._speak_weather(dialog)
+
+    def _display_current_weather(self, report):
+        image_code = self.CODES[report.current.condition.icon]
+        if self.gui.connected:
+            self.gui["current"] = report.current.temperature
+            self.gui["min"] = report.current.low_temperature
+            self.gui["max"] = report.current.high_temperature
+            self.gui["condition"] = report.current.condition.description
+            self.gui["icon"] = report.current.condition.icon
+            self.gui["weathercode"] = image_code
+            self.gui.show_pages(["weather.qml", "highlow.qml"])
```

Comment:
Is the removal of the transition to the high/low page intentional?
Also the forecast weather pages aren't loaded. Is that intentional as well?

Reply:
The GUI will be my second pass.  Just wanted to get the intents working and some xfail VK tests passing.